### PR TITLE
connecting Vis5, vis6 and vis7 with backend

### PR DESCRIPTION
**Pull request by MohHajri**

### CODE REVIEW

File: src/components/charts/Vis5.js [hunk 1]
```diff
@@ -57,20 +67,33 @@ export const options = {
...
+  useEffect(() => {
     setDatasets(generateCo2Data(data))
   }, [data])
 
   function generateCo2Data(data) {
     const cO2Datasets = [
       {
         label: 'Vostok Ice Core CO2 measurements',
-        data: data.filter((item) => item['CO2 concentration']),
+        data: data.filter((item) => item['y']),
```
Comment (Rexol) [inline src/components/charts/Vis5.js:96]: This line does nothing

File: src/components/charts/Vis6.js [hunk 2]
```diff
@@ -56,21 +67,33 @@ export const options = {
...
   const [datasets, setDatasets] = useState([])
 
   useEffect(() => {
-    setData(DATA)
+    axios
+      .get(URL)
+      .then((response) => {
+        let data = response.data.data
+        setData(data)
+        setDatasets(generateIceCoreData(data))
```
Comment (Rexol) [inline src/components/charts/Vis6.js:81]: This line won't work, since executed right after set data. Put it into other useeffect

Comment (Rexol) [inline src/components/charts/Vis6.js:81]: This line won't work as expected

File: src/components/charts/Vis6.js [hunk 3]
```diff
@@ -56,21 +67,33 @@ export const options = {
...
+  useEffect(() => {
     setDatasets(generateIceCoreData(data))
   }, [data])
 
   function generateIceCoreData(data) {
     const iceCoreDatasets = [
       {
         label: 'Ice core 800k year composite study CO2 measurements',
-        data: data.filter((item) => item['CO2 concentration (ppm)']),
+        data: data.filter((item) => item['y']),
```
Comment (Rexol) [inline src/components/charts/Vis6.js:96]: This line does nothing

File: src/components/charts/Vis7.js [hunk 4]
```diff
@@ -81,27 +80,37 @@ export const options = {
...
+        })
+      )
+      .catch((e) => {
+        alert(e)
+      })
+  }, [])
+
+  useEffect(() => {
     generateAnnualData()
   }, [data])
```
Comment (Rexol) [inline src/components/charts/Vis7.js:109]: This useffect will be always executing -> slow
Why? You set data inside it and it is automatically executed, when data changed so you set data, call useeffect, set data, cal useffect, setdata. . .

File: src/components/charts/Vis7.js [hunk 5]
```diff
@@ -81,27 +80,37 @@ export const options = {
...
+    const req1 = axios.get(URL1)
+    const req2 = axios.get(URL2)
+    axios
+      .all([req1, req2])
+      .then(
+        axios.spread((...response) => {
+          let data1 = response[0].data.data
+          let data2 = response[1].data.data
+          setData({ tempData: data1, co2Data: data2 })
+          generateAnnualData()
```
Comment (Rexol) [inline src/components/charts/Vis7.js:99]: Why do you call this function 2 times in useffect?

**Review by Rexol — CHANGES_REQUESTED** [review]

**Comment by Rexol** [comment]
Pull main branch with rebase, because you have also problem with commits

File: src/components/charts/Vis5.js (hunk 1 above)
Comment (MohHajri) [reply src/components/charts/Vis5.js]: done

File: src/components/charts/Vis6.js (hunk 3 above)
Comment (MohHajri) [reply src/components/charts/Vis6.js]: done

File: src/components/charts/Vis7.js (hunk 5 above)
Comment (MohHajri) [reply src/components/charts/Vis7.js]: done

File: src/components/charts/Vis6.js (hunk 2 above)
Comment (MohHajri) [reply src/components/charts/Vis6.js]: done

**Comment by Rexol** [comment]
Have you tried `git pull --rebase origin main`? You still have commits **add vis7** and **merge branch...** that should not go to master
If pull with rebase hasn't helped, then you can use
```
git reset --soft HEAD~3
git commit -m ""
```
to get only 1 commit for pull request

**Comment by MohHajri** [comment]
done

**Comment by Rexol** [comment]
Why are you trying to commit vis1, vis3, vis8 files? restrore soft to HEAD~1 again and add only files that needs to be commited

**Review by Rexol — APPROVED** [review]

**Review by ni9iri — APPROVED** [review]

**Pull request merged**